tensorflow/core/kernels/data/name_utils.cc

```diff
@@ -41,15 +41,23 @@ string OpName(const string& dataset_type) {
     return "PrefetchDataset";
   } else if (dataset_type == "Range") {
     return "RangeDataset";
+  } else if (dataset_type == "Shard") {
+    return "ShardDataset";
+  } else if (dataset_type == "Shuffle") {
+    return "ShuffleDataset";
+  } else if (dataset_type == "ShuffleAndRepeat") {
+    return "ShuffleAndRepeatDataset";
   }
   LOG(WARNING) << "Unknown dataset type " << dataset_type << std::endl;
   return "UnknownDataset";
 }
 
 string DatasetDebugString(const string& dataset_type,
+                          const string& dataset_name,
```

Comment:
I would prefer if this is `dataset_name_prefix` that defaults to "" in the overload that does not expose the argument?

Reply:
Good suggestion. Changed it.